docs/verifiable-document/document-template.md

```diff
@@ -4,8 +4,228 @@ title: Deploying Document Renderer
+
+### Registering the COC template
+
+`src/templates/index.tsx` is a directory of all the templates available in this renderer.
+
+To register a new template, simply add it as a key to the `registry` constant. Take note that the key will be the `$template.name` value defined in the document data.
+
+Replace `src/templates/index.tsx` with the following code to add the new `COC` template:
+
+```js
+import { TemplateRegistry } from "@govtechsg/decentralized-renderer-react-components";
+import { templates } from "./customTemplate";
+
+export const registry: TemplateRegistry<any> = {
+  default: templates,
+  COC: templates
+};
+```
+
+*Note: The `default` template is a special template that will be used as the fallback template when a template of the correct name cannot be found.*
+
+### Multiple views for a Template
+
+An OA document may have multiple views, each of them rendered in a separate tabs. The views are defined in the `templates` array in `index.tsx` of the template folder. Each of these views can have its own id, label and template code.
+
+For our Certificate of Completion, we will only use a single view. So we will remove any additional views and rename the first view's label as "Certificate".
+
+Replace the code in `src/templates/customTemplate/index.tsx` with the following:
+
+```js
+import { CustomTemplate } from "./customTemplate";
+
+export const templates = [
+  {
+    id: "certificate",
+    label: "Certificate",
+    template: CustomTemplate
+  }
+];
+```
+
+### Developing the COC Template View
+
+Finally, once all the components have been wired up, we may proceed to style our Certificate of Completion.
+
+To change how the data is being renderered, we simply create a React component that takes in the unwrapped document in the `document` props and render the corresponding HTML code.
+
+For our COC, we will simply display the following text:
+
+```text
+OpenAttestation Tutorial Certificate of Completion
+awarded to
+John Doe
+```
+
+To do so, you may update the template code in `src/templates/customTemplate/customTemplate.tsx` to the following:
+
+```js
+import React, { FunctionComponent } from "react";
+import { TemplateProps } from "@govtechsg/decentralized-renderer-react-components";
+import { css } from "@emotion/core";
+import { CustomTemplateCertificate } from "../sample";
+import { PrintWatermark } from "../../core/PrintWatermark";
+
+const containerStyle = css`
+  background-color: #324353;
+  color: #ffffff;
+  padding: 15px;
+  margin: auto;
+  width: 80%;
+  text-align: center;
+`;
+
+export const CustomTemplate: FunctionComponent<
+  TemplateProps<CustomTemplateCertificate> & { className?: string }
+> = ({ document, className = "" }) => {
+  return (
+    <div css={containerStyle} className={className} id="custom-template">
+      <PrintWatermark />
+      <h1>{document.name}</h1>
+      <div>awarded to</div>
+      <h2>{document.recipient.name}</h2>
+    </div>
+  );
+};
+```
+
```

Comment:
Should we add a final note that for they can change the name of the templates / variables / types ... the way they want (it doesnt amtter as long as the template configuration is valid)

Reply:
Dont understand this one here... Do you mean to say that they are free to name the template? Think that is covered in the key section. However if it's about variable naming, I dont want to talk about that because they shouldnt be reading this document if they cannot even program.